rest/route/agent.go

```diff
@@ -1823,14 +1823,14 @@ func (h *awsS3Credentials) Run(ctx context.Context) gimlet.Responder {
 	}
 	policyJSON, err := json.Marshal(sessionPolicy)
 	if err != nil {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "marshalling session policy for task '%s'", h.taskID))
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "marshalling session policy for task '%s'", h.taskID))
```

Comment:
While I'm gonna guess that this error case will never happen, this is probably a genuine 500 if it does occur, since the marshalling logic is entirely within the app server and we're not already retrying a request on behalf of the user to an external service (i.e. AWS).

Reply:
That's sound logic, I was thinking the flip side where if this did error then retrying probably won't help but I think a 500 makes more sense since it's a genuine 500!